[BUG]

**Describe the bug**
In the spirit of the discussion during the ASWF meeting, making option things optional :)

Even if tests are disabled (`OIIO_BUILD_TESTS=FALSE`) CMake still tries to clone oiio-tests.

**To Reproduce**
Steps to reproduce the behavior:
1. Add -DOIIO_BUILD_TEST=FALSE option to CMake
2. Run CMake
3. Then THIS happens:
```
-- Cloning oiio-images from https://github.com/OpenImageIO/oiio-images.git
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Git (missing: GIT_EXECUTABLE)
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake/Modules/FindGit.cmake:128 (find_package_handle_standard_args)
  src/cmake/testing.cmake:346 (find_package)
  src/cmake/testing.cmake:357 (oiio_get_test_data)
  CMakeLists.txt:289 (oiio_setup_test_data)
```

**Expected behavior**
All the test related CMake commands should be wrapped in a conditional not to execute if testing is not enabled/used.

**Platform information:**
 - OIIO branch/version: 2.3.10
 - OS: Fedora Linux
 - C++ compiler: 11.2.1
 - Any non-default build flags when you build OIIO:
N/A

Probably update the existing conditional for PROJECT_IS TOP_LEVEL?

--- a/src/cmake/testing.cmake
+++ b/src/cmake/testing.cmake
@@ -343,10 +343,14 @@ function (oiio_get_test_data name)
             if (NOT _ogtd_BRANCH)
                 set (_ogtd_BRANCH master)
             endif ()
-            find_package (Git REQUIRED)
-            execute_process(COMMAND ${GIT_EXECUTABLE} clone --depth 1
-                                    ${_ogtd_REPO} -b ${_ogtd_BRANCH}
-                                    ${CMAKE_BINARY_DIR}/testsuite/${name})
+            find_package (Git)
+            if (Git_FOUND AND GIT_EXECUTABLE)
+                execute_process(COMMAND ${GIT_EXECUTABLE} clone --depth 1
+                                        ${_ogtd_REPO} -b ${_ogtd_BRANCH}
+                                        ${CMAKE_BINARY_DIR}/testsuite/${name})
+            else ()
+                message (WARNING "Could not find Git executable, could not download test data from ${_ogtd_REPO}")
+            endif ()
         endif ()
     else ()
         message (STATUS "Missing test data ${name}")

